### PR TITLE
Save ticket with save guest user

### PR DESCRIPTION
**Pull request by 93RakeshS**
with parameter Dictionary<string,object> data where string is UserModel/TicketModel and object is respective object,
PS. this branch is created from refactoring branch

### CODE REVIEW

File: TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Controllers/TicketBookingController.cs [hunk 1]
```diff
@@ -46,10 +48,31 @@ public async Task<ActionResult<BusModel>> GetBus(int busId)
...
             var res = await _ticketService.SaveTicket(ticketModel);
             return Ok(res);
         }
+
+        [HttpPost("SaveTicketForGuest")]
+        public async Task<ActionResult<bool>> SaveTicketForGuest(Dictionary<string, object> data)
+        {
+            bool res = false;
+            int userId = 0;
+            var userModel = (UserModel)data["UserModel"];
```
Comment (ViswanthSwarna) [inline TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Controllers/TicketBookingController.cs:62]: move this to constants file

File: TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Controllers/TicketBookingController.cs [hunk 2]
```diff
@@ -46,10 +48,31 @@ public async Task<ActionResult<BusModel>> GetBus(int busId)
...
         [HttpPost("SaveTicket")]
-        public async Task<ActionResult<bool>> SaveTicket(TicketModel ticketModel) 
+        public async Task<ActionResult<bool>> SaveTicket(TicketModel ticketModel)
         {
             var res = await _ticketService.SaveTicket(ticketModel);
             return Ok(res);
         }
+
+        [HttpPost("SaveTicketForGuest")]
+        public async Task<ActionResult<bool>> SaveTicketForGuest(Dictionary<string, object> data)
```
Comment (ViswanthSwarna) [inline TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Controllers/TicketBookingController.cs:58]: divide this into multiple functions for better readability

File: TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Repository/UserRepository.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+
+        public async Task<int> SaveUser(UserModel userModel) 
+        {
+            var user = _mapper.Map<ActiveUser>(userModel);
+            _context.Entry(user).State = EntityState.Added;
+            var res = await _context.SaveChangesAsync();
```
Comment (rahul-calvi) [inline TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Repository/UserRepository.cs:24]: remove unused variable, and directly return user.Id

File: TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Controllers/TicketBookingController.cs (hunk 2 above)
Comment (rahul-calvi) [inline TicketBookingBackend/TicketBookingAPI/TicketBookingAPI/Controllers/TicketBookingController.cs:58]: This in not correct approach, also it not very clear what it is storing in dictionary

**Pull request merged**